org.eclipse.xtext.builder.tests/src/org/eclipse/xtext/builder/impl/AbstractParticipatingBuilderTest.java

```diff
@@ -44,35 +44,35 @@ public void tearDown() throws Exception {
 	}
 	
 	@Override
-	public void build(IBuildContext context, IProgressMonitor monitor) throws CoreException {
+	public synchronized void build(IBuildContext context, IProgressMonitor monitor) throws CoreException {
```

Comment:
This is usually executed in a job / workspace lock. Why does it help to make it synchronized?